src/AWS.jl

```diff
@@ -27,7 +28,7 @@ using ..AWSExceptions
 using ..AWSExceptions: AWSException
 
 user_agent = Ref("AWS.jl/1.0.0")
-aws_config = Ref{AWSConfig}()
+aws_config = Ref{AbstractAWSConfig}()
```

Comment:
These should be defined as constants

Reply:
These are `Refs` because we have the option to change them for default behaviours using the `global_aws_config()` functions.

Reply:
I was unclear. These should stay using `Ref` but the variable should be constant:
```suggestion
const aws_config = Ref{AbstractAWSConfig}()
```
You can still change the contents of the `Ref` but the the `aws_config` will always be a `Ref{AbstractAWSConfig}`. This helps the compiler during inference.

Reply:
Ah I misunderstood what you were asking here